ui/jobs/jobs.js

```diff
@@ -1075,13 +1075,20 @@ class Bars {
 
     this.contentType = 0;
 
+    this.crafting = false;
+    this.craftingType = '';
+
     const lang = this.options.ParserLanguage;
     this.countdownStartRegex = LocaleRegex.countdownStart[lang] || LocaleRegex.countdownStart['en'];
     this.countdownCancelRegex = LocaleRegex.countdownCancel[lang] || LocaleRegex.countdownCancel['en'];
     this.craftingStartRegex = LocaleRegex.craftingStart[lang] || LocaleRegex.craftingStart['en'];
     this.craftingFinishRegex = LocaleRegex.craftingFinish[lang] || LocaleRegex.craftingFinish['en'];
     this.craftingFailRegex = LocaleRegex.craftingFail[lang] || LocaleRegex.craftingFail['en'];
     this.craftingCancelRegex = LocaleRegex.craftingCancel[lang] || LocaleRegex.craftingCancel['en'];
+    this.trialCraftingStartRegex = LocaleRegex.trialCraftingStart[lang] || LocaleRegex.trialCraftingStart['en'];
+    this.trialCraftingFinishRegex = LocaleRegex.trialCraftingFinish[lang] || LocaleRegex.trialCraftingFinish['en'];
+    this.trialCraftingFailRegex = LocaleRegex.trialCraftingFail[lang] || LocaleRegex.trialCraftingFail['en'];
+    this.trialCraftingCancelRegex = LocaleRegex.trialCraftingCancel[lang] || LocaleRegex.trialCraftingCancel['en'];
```

Comment:
There's a lot of crafting regexes now here.  What about combining them so that the logic below is easier? e.g.
```
const trialCraftingStartRegex = ...
const trialCraftingFinishRegex = ...
...

this.craftingStartRegex = Regexes.anyOf(craftingStartRegex, trialCraftingStartRegex);
this.craftingStopRegex = Regexes.anyOf(craftingFailRegex, craftingCancelRegex, trialCraftingFailRegex, trialCraftingCancelRegex);
```

Then you can just check against `this.craftingStartRegex` and `this.craftingStopRegex` below, which should make that logic simpler.

Reply:
Yeah. But `Regexes.anyOf` just combine them with `(?:exp1|exp2)`, it will cause *SyntaxError* when there are named capture groups in all the regexes.
![image](https://user-images.githubusercontent.com/19927330/93692309-00db7c00-fb24-11ea-9da9-6d6a7bdaec28.png)

Reply:
Oh, whoops.  Sorry about that suggestion, then.  I think the way you've added a helper function makes this more readable.